cp23/hpi-fun2/motivation/executables/generator.py

```diff
@@ -0,0 +1,232 @@
+
+
+def negative_root_node_weight():
+    filename = "negative_root_node_weight"
+
+    test_case = (
+        "5 10\n"
+        "-100 90 100 110 120\n"
+        "1 1 1 1"
+    )
+    Path(filename + ".in").write_text(test_case)
+    Path(filename + ".desc").write_text("Root has negative weight and children are partially reachable.")
+
+
+def binary_tree_two_layers():
+    filename = "binary_tree_two_layers"
+
+    test_case = (
+        "7 10\n"
+        "5 5 5 1 1 1 1\n"
+        "1 1 3 3 2 2"
+    )
+    Path(filename + ".in").write_text(test_case)
+    Path(filename + ".desc").write_text("Binary tree of 3 layers with unreachable third layer.")
+
+
+def unreachable_subtree_with_negatives():
+    filename = "unreachable_subtree_with_negatives"
+
+    test_case = (
+        "11 20\n"
+        "1 3 7 -2 2 -6 1 30 3 5 5\n"
+        "1 1 1 3 3 3 4 4 8 8"
+    )
+    Path(filename + ".in").write_text(test_case)
+    Path(filename + ".desc").write_text("Unreachable subtree in a tree including negative weights.")
+
+
+def exponential_weights():
+    filename = "exponential_weights"
+
+    test_case = (
+        "14 15\n"
+        "1 2 2 2 4 4 4 4 8 8 8 8 16 16\n"
+        "1 1 1 3 3 4 4 6 6 8 8 10 10"
+    )
+    Path(filename + ".in").write_text(test_case)
+    Path(filename + ".desc").write_text("Exponentially increasing weights of simple tree.")
+
+
+def single_node_bottleneck():
+    filename = "single_node_bottleneck"
+
+    test_case = (
+        "6 5\n"
+        "1 10 -8 -8 -8 -8\n"
+        "1 2 2 2 2"
+    )
+    Path(filename + ".in").write_text(test_case)
+    Path(filename + ".desc").write_text("There is a single bottleneck in the tree, which renders all its children unreachable.")
+
+
+def large_increasing_chain():
+    test_case = ""
+
+    employees = INPUT_LIMIT
+    threshold = INPUT_LIMIT // 2
+    test_case += f"{employees} {threshold}\n"
+
+    levels = []
+    for i in range(0, employees):
+        levels.append(1)
+    test_case += " ".join([str(x) for x in levels]) + "\n"
+
+    bosses = []
+    for i in range(1, employees):
+        bosses.append(i)
+    test_case += " ".join([str(x) for x in bosses])
+
+    filename = "large_increasing_chain"
+    Path(filename + ".in").write_text(test_case)
+    Path(filename + ".desc").write_text("A long chain (only single child per node), where each child was weight 1.\nLimit Test.")
+
+
+def switching_chain():
+    test_case = ""
+
+    employees = 1_000
+    threshold = 1
+    test_case += f"{employees} {threshold}\n"
+
+    levels = []
+    for i in range(0, employees):
+        if i % 2 == 0:
+            levels.append(1)
+        else:
+            levels.append(-1)
+    test_case += " ".join([str(x) for x in levels]) + "\n"
+
+    bosses = []
+    for i in range(1, employees):
+        bosses.append(i)
+        test_case += f"{i} "
+    test_case += " ".join([str(x) for x in bosses]) + "\n"
```

Comment:
```suggestion
    test_case += " ".join(str(x) for x in range(1, employees)) + "\n"
```
also the current code generates too much input, right?

Reply:
Yes, I guess something broke during merging.